front-end/src/components/analysis-page/IntentDiagram.jsx

```diff
@@ -1,17 +1,52 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import store from '../../store.js';
-import { toggleFlag, toggleStar } from '../../store/analyzeTranscriptSlice.js';
+import { goToIntentByQuestion, toggleFlag, toggleStar } from '../../store/analyzeTranscriptSlice.js';
 import StarAPI from "../../services/StarAPI.js";
 import FlagAPI from "../../services/FlagAPI.js";
 
 class IntentDiagram extends Component {
+
+    generateIntentBoxStyling = (flash) => {
+        // Base styling
+        let styling = "rounded-lg p-5 mt-9 shadow-lg shadow-blue/10 break-words text-center text-lg 2xl:text-2xl w-52 sm:min-w-[30rem] sm:max-w-[35rem] 2xl:max-h-[14rem] ";
+        // Add color based on whether the box is flashing
+        styling += flash ? "bg-green-100" : "bg-off-white";
+        // Add any previous intents merged into this intent, if any
+        if (this.props.previousIntents) {
+
+        }
+        return styling
+    }
+
     listLeaves = () => {
         return Object.keys(this.props.children).map((key) => {
+            const isEndOfConversation = (key === "END OF CONVERSATION");
+
+            const goToLeaf = () => {
+                store.dispatch(goToIntentByQuestion(key))
+                // Reset focus on clicked leaf (it doesn't do it automatically)
+                document.activeElement.blur()
+                // Find the new intent question box by id, then focus and flash it to highlight its position
+                setTimeout(() => { document.getElementById(key).focus() }, 0);
+                setTimeout(() => { document.getElementById(key).className = this.generateIntentBoxStyling(true) }, 0);
+                setTimeout(() => { document.getElementById(key).className = this.generateIntentBoxStyling(false) }, 150);
+                setTimeout(() => { document.getElementById(key).className = this.generateIntentBoxStyling(true) }, 300);
+                setTimeout(() => { document.getElementById(key).className = this.generateIntentBoxStyling(false) }, 450);
+            }
+
             return (
                 <div 
-                    className="text-md bg-green-100 shadow-md shadow-blue/10 rounded-2xl pb-2 mb-5 border-2 border-green-200"
+                    className={"text-md bg-green-200 shadow-md shadow-blue/10 rounded-2xl pb-2 mb-5 border-2 border-green-300 " +
```

Comment:
![image](https://user-images.githubusercontent.com/76976842/205464391-2207f31d-55e7-44a2-9835-1849de2c7d1c.png)
Can we make the leaf have a red BG when EOC?